templates/tfengine/components/iam_bindings/main.tf

```diff
@@ -0,0 +1,65 @@
+{{- /* Copyright 2021 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */ -}}
+
+{{range $index, $value := get . "iam_bindings"}}
+module "{{resourceName . "parent_type"}}_iam_bindings_{{$index}}" {
+  {{- if eq .parent_type "storage_bucket"}}
+  source = "terraform-google-modules/iam/google//modules/storage_buckets_iam"
+  storage_buckets = [
+    {{- range .parent_ids}}
+    "{{.}}",
+    {{- end}}
+  ]
+  {{- else if eq .parent_type "project"}}
+  source = "terraform-google-modules/iam/google//modules/projects_iam"
+  projects = [
+    {{- range .parent_ids}}
+    "{{.}}",
+    {{- end}}
+  ]
+  {{- else if eq .parent_type "organization"}}
+  source = "terraform-google-modules/iam/google//modules/organizations_iam"
+  organizations = [
+    {{- range .parent_ids}}
+    "{{.}}",
+    {{- end}}
+  ]
+  {{- else if eq .parent_type "folder"}}
```

Comment:
for folder, should the parent id be the folder name or ID? I think it probably should be the ID. Could you test that? It's not clear in https://github.com/terraform-google-modules/terraform-google-iam/tree/master/modules/folders_iam and it seems using folder name there which I think might be incorrect.

And usually for folders, the folder id should be prefixed with `folders/`.

Reply:
I did test this. It should be the folder ID, but it doesn't require the `folders/` prefix.